scripts/proper_bs.py

```diff
@@ -0,0 +1,100 @@
+from run_sweep import WORKER_TIMEOUT, WARMUP_ROUNDS, ModelTestResult, NANOSECONDS_PER_MILLISECONDS
+
+
+def run_one_step_flops(func, device: str, nwarmup=WARMUP_ROUNDS, num_iter=10, flops=True) -> Tuple[float, float, Optional[Tuple[torch.Tensor]]]:
+    "Run one step of the model, and return the latency in milliseconds."
+    # Warm-up `nwarmup` rounds
+    for _i in range(nwarmup):
+        func()
+    result_summary = []
+    if flops:
+        model_analyzer = ModelAnalyzer()
+        model_analyzer.start_monitor()
+    for _i in range(num_iter):
+        if device == "cuda":
+            torch.cuda.synchronize()
+            # Collect time_ns() instead of time() which does not provide better precision than 1
+            # second according to https://docs.python.org/3/library/time.html#time.time.
+            t0 = time.time_ns()
+            func()
+            torch.cuda.synchronize()  # Wait for the events to be recorded!
+            t1 = time.time_ns()
+        else:
+            t0 = time.time_ns()
+            func()
+            t1 = time.time_ns()
+        result_summary.append((t1 - t0) / NANOSECONDS_PER_MILLISECONDS)
+    if flops:
+        model_analyzer.stop_monitor()
+        model_analyzer.aggregate()
+        tflops = model_analyzer.calculate_flops()
+        
+    wall_latency = numpy.median(result_summary)
+    return (wall_latency, tflops)
+
+def _run_model_test_proper_bs(model_path: pathlib.Path, test: str, device: str, jit: bool, batch_size: Optional[int], extra_args: List[str]) -> ModelTestResult:
+    assert test == "train" or test == "eval", f"Test must be either 'train' or 'eval', but get {test}."
+    result = ModelTestResult(name=model_path.name, test=test, device=device, extra_args=extra_args, batch_size=None, precision="fp32",
+                             status="OK", results={})
+
+    # Run the benchmark test in a separate process
+    print(f"Running model {model_path.name} ... ", flush=True)
+    status: str = "OK"
+    bs_name = "batch_size"
+    correctness_name = "correctness"
+    error_message: Optional[str] = None
+    result.results['details'] = []
+    task = ModelTask(os.path.basename(model_path), timeout=WORKER_TIMEOUT)
+    for batch_size in range(1, 128):
+        try:
+            print(f"Batch Size {batch_size} ", end='')
+            latency_ms_cur = 0
+            if not task.model_details.exists:
+                status = "NotExist"
+                return
+            task.make_model_instance(test=test, device=device, jit=jit, batch_size=batch_size, extra_args=extra_args)
+            result.precision = task.get_model_attribute("dargs", "precision")
+            # Check the batch size in the model matches the specified value
+            if batch_size and (not task.get_model_attribute(bs_name) == batch_size):
+                raise ValueError(f"User specify batch size {batch_size}, but model {result.name} runs with batch size {task.get_model_attribute(bs_name)}. Please report a bug.")
```

Comment:
We need to update this message: some models don't support tuning batch size. For those models, we will set the attribute `ALLOW_CUSTOMIZE_BSIZE = False` (https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/drq/__init__.py#L90), so you should first check if the model has this attribute and it is set to False. If so, raise a ValueError exception with message "Model does not support tuning batch size", and don't need to run the test.